pkg/cmd/container/create.go

```diff
@@ -357,8 +365,49 @@ func generateRootfsOpts(args []string, id string, ensured *imgutil.EnsuredImage,
 			// error message is from Podman
 			return nil, nil, errors.New("no command or entrypoint provided, and no CMD or ENTRYPOINT from image")
 		}
+
+		entrypointPath = processArgs[0]
+
 		opts = append(opts, oci.WithProcessArgs(processArgs...))
 	}
+
+	systemdPaths := []string{
+		"/usr/sbin/init",
+		"/sbin/init",
+		"/usr/local/sbin/init",
+	}
+
+	stopSignal := options.StopSignal
+
+	if options.Systemd == "always" || (options.Systemd == "true" && slices.Contains(systemdPaths, entrypointPath)) {
+		if options.Privileged {
+			securityOptsMap := strutil.ConvertKVStringsToMap(strutil.DedupeStrSlice(options.SecurityOpt))
+			privilegedWithoutHostDevices, err := maputil.MapBoolValueAsOpt(securityOptsMap, "privileged-without-host-devices")
+			if err != nil {
+				return nil, nil, err
+			}
+
+			// See: https://github.com/containers/podman/issues/15878
+			if !privilegedWithoutHostDevices {
+				return nil, nil, errors.New("If --privileged is used with systemd `--security-opt privileged-without-host-devices` must also be used")
+			}
+		}
+
+		opts = append(opts,
+			oci.WithoutMounts("/sys/fs/cgroup"),
+			oci.WithMounts([]specs.Mount{
+				{Type: "cgroup", Source: "cgroup", Destination: "/sys/fs/cgroup", Options: []string{"rw"}},
+				{Type: "tmpfs", Source: "tmpfs", Destination: "/run"},
+				{Type: "tmpfs", Source: "tmpfs", Destination: "/run/lock"},
+				{Type: "tmpfs", Source: "tmpfs", Destination: "/tmp"},
+				{Type: "tmpfs", Source: "tmpfs", Destination: "/var/lib/journal"},
+			}),
+		)
+		stopSignal = "SIGRTMIN+3"
```

Comment:
`SIGTERM` causes restart in `systemd` (This functionality is the same as podman)

See:
https://www.freedesktop.org/software/systemd/man/latest/systemd.html#Signals